### DISCUSSION TITLE
Why uppercase properties generated within setter and getter? - lowercase required

### DISCUSSION THREAD

**Discussion by trailsnail** [phpro/soap-client#396] [Q&A]
Hi again,

I have an WSDL file with two scheme files.
`cis_bas.xsd` and `bcs_base.xsd`

The generated class contains lowercase properties and the same as uppercase variants.
For the uppercase variants the getter and setter are built.

This causes error since endpoint await that the lower case stuff is filled.

So how can I avoid this? 

![grafik](https://user-images.githubusercontent.com/4742603/143449439-50d5a6e9-076d-45de-a67e-4e58bff485cf.png)

**Comment by veewee** [comment]
The getter and setter normalizes the name of the properties as get|set.ucfirst(prop)
If the method already exists, it is either patched or skipped.

That is why you are seeing this.

You can overcome this issue by creating your own get and set assembler, which has a different normalization strategy or which tries to read/write both properties.

However, i. This case the real issue is the fact that both complex types get merged because they have the same name and because ext-soap is not able to do class map matching on multiple namespaces.

If the case of the props in the service were consise, this wouldn't be a problem.
